### PR TITLE
Bug 1813062: Updates Status Reconciliation to Support DNS Service

### PR DESCRIPTION
**Pull request by danehans**
Previously, the operator would surface status conditions indicating an issue with the DNS daemonset if the DNS service could not be created. This PR does the following:

1. Updates the Makefile to add support for running the operator locally.
2. Updates the Makefile so the test-e2e target only runs e2e tests.
3. Creates a separate controller for status (similar to ingress operator). Operator status is now based on the availability of `dns` instead of dependent resources of `dns`.
4. Updates unit tests.
5. Updates e2e tests.

/assign @Miciah @knobunc 
/cc @frobware @sgreene570

### CODE REVIEW

File: test/e2e/utils.go [hunk 1]
```diff
@@ -228,3 +238,54 @@ func buildContainer(name, image string, cmd []string) corev1.Container {
...
+		current := clusterOperatorConditionMap(co.Status.Conditions...)
+		return conditionsMatchExpected(expected, current), nil
+	})
+}
+
+func waitForDNSConditions(cl client.Client, timeout time.Duration, name types.NamespacedName, conditions ...operatorv1.OperatorCondition) error {
+	return wait.PollImmediate(1*time.Second, timeout, func() (bool, error) {
+		dns := &operatorv1.DNS{}
+		if err := cl.Get(context.TODO(), name, dns); err != nil {
+			return false, err
```
[the remark this answers is not in the document]
Comment (danehans) [reply test/e2e/utils.go:259]: @sgreene570 L259 mirrors ingress operator [here](https://github.com/openshift/cluster-ingress-operator/blob/master/test/e2e/operator_test.go#L1802). I prefer the 2 operators are as consistent and address both in a follow-on PR.

**Comment by danehans** [comment]
`e2e-aws` job failed due to exceeding aws throttling rate:
```
2020-08-19T21:16:05.721092959Z 2020-08-19T21:16:05.721Z	ERROR	operator.dns_controller	dns/controller.go:181	failed to publish DNS record to zone	{"record": {"dnsName":"*.apps.ci-op-x0729h3t-bcd5b.origin-ci-int-aws.dev.rhcloud.com.","targets":["a767b6002f7ce4760a6e2bb4fe68426a-899735716.us-east-1.elb.amazonaws.com"],"recordType":"CNAME","recordTTL":30}, "dnszone": {"tags":{"Name":"ci-op-x0729h3t-bcd5b-p7cq2-int","kubernetes.io/cluster/ci-op-x0729h3t-bcd5b-p7cq2":"owned"}}, "error": "failed to update alias in zone Z04249512EGDZ53SW9JG0: couldn't update DNS record in zone Z04249512EGDZ53SW9JG0: Throttling: Rate exceeded\n\tstatus code: 400, request id: 3d698c07-3d1c-41cf-b666-bcc72d7fde71"}
```

The good thing is that the operator surfaced a `Degraded=True` condition. I don't see how this PR causes the throttling issue.

/test e2e-aws

**Comment by sgreene570** [comment]
CI is green.

/lgtm

**Comment by sgreene570** [comment]
```
"Pod openshift-marketplace/community-operators-f7gt8 is not healthy: Back-off pulling image \"quay.io/openshift-community-operators/catalog:latest\"",
```
/retest

**Pull request merged**